### ISSUE TITLE
autorenew is not working

### ISSUE THREAD

**Issue by mbrde** [certbot/certbot#9931]
Hi,

i have certbot 1.22.0 running on oracle linux 8:

certbot.noarch                       1.22.0-1.el8                                  @ol8_developer_EPEL
python3-certbot.noarch               1.22.0-1.el8                                  @ol8_developer_EPEL
python3-certbot-apache.noarch        1.22.0-1.el8                                  @ol8_developer_EPEL

after creating the certificate with "certbot certonly --apache -d DOMAIN" it's confirmed with the message "Certbot has set up a scheduled task to automatically renew this certificate in the background." and everything works, but there is no background job running.

is there an issue in this specific release or am i missing something else? I hope the answer is not "use the snap package".

**Comment by osirisinferi** [comment]
> but there is no background job running

What exactly do you mean by this? How/where did you check?

**Comment by mbrde** [comment]
> > but there is no background job running
> 
> What exactly do you mean by this? How/where did you check?

nothing happened:
- still the old certs in /etc/letsencrypt/live
- no new entry in /var/log/letsencrypt

also i'm not sure how the task is triggered, but there is neither a cronjob nor a systemd-timer.

**Comment by osirisinferi** [comment]
I don't fully understand why you would see that message. It would only show if the `--preconfigured-renewal` option was used, which is a packager use only option (i.e., not to be used by users on the command line, but by the packager of an OS package).

If I take a look at the contents of the [OL8 EPEL Certbot 1.22.0-1 RPM](https://yum.oracle.com/repo/OracleLinux/OL8/developer/EPEL/x86_64/getPackage/python3-certbot-1.22.0-1.el8.noarch.rpm), I do not see such option *actually* being used anywhere.

That said, I also can't find that option in the Debian `.deb` package of which I know it does install a systemd timer.. :thinking: So maybe I'm not searching good enough.

Anyway, Certbot *itself* (the Python application) does not automatically insert cronjobs or systemd timers and relies on the `--preconfigured-renewal` option to show you that message, which is added by packagers. The recommended method of installing Certbot using `snap` however does install a (indirect I believe) systemd timer. For OS packages it's left to the packager of that package to add a systemd timer or cronjob.

So while I don't fully understand the OL8 EPEL package, I don't believe this is actually a Certbot issue.

**Comment by mbrde** [comment]
> I don't fully understand why you would see that message. It would only show if the `--preconfigured-renewal` option was used, which is a packager use only option (i.e., not to be used by users on the command line, but by the packager of an OS package).

Well, I see the message.

> So while I don't fully understand the OL8 EPEL package, I don't believe this is actually a Certbot issue.

At least certbot shows the wrong message.

> Anyway, Certbot _itself_ (the Python application) does not automatically insert cronjobs or systemd timers and relies on the `--preconfigured-renewal` option to show you that message, which is added by packagers. The recommended method of installing Certbot using `snap` however does install a (indirect I believe) systemd timer. For OS packages it's left to the packager of that package to add a systemd timer or cronjob.

I never understood why snap is the recommended method. Does anybody really like snap?

Back to topic: For now I will set up a systemd timer or a cronjob manually in future. But maybe someone finds out what is really going wrong here. Either Oracle Linux (or RHEL) made a mistake in the EPEL Repo or I missed something...

**Comment by osirisinferi** [comment]
> At least certbot shows the wrong message.

No, Certbot shows a message as instructed by the `--preconfigured-renewal` option. Of course I cannot exclude a bug with the limited information provided, but I highly doubt it. If you would have shown a log as requested by the initial questionnaire when you opened this issue (which you have deleted), we could investigate further.

> But maybe someone finds out what is really going wrong here. Either Oracle Linux (or RHEL) made a mistake in the EPEL Repo or I missed something...

You might want to reach out to the OL8 EPEL packager.

**Comment by mbrde** [comment]
> No, Certbot shows a message as instructed by the `--preconfigured-renewal` option.

You are right. What I meant was that this message made me think, certbot creates the timer.

> Of course I cannot exclude a bug with the limited information provided, but I highly doubt it. If you would have shown a log as requested by the initial questionnaire when you opened this issue (which you have deleted), we could investigate further.

Sorry for the few information. This issue is reproducable on a fresh OL8 with only certbot and httpd or nginx installed.
I have checked all logfiles. After creating the certificates no new entry was appended and while creating no error occoured. But I will check the logs again and provide them here soon. 

While answering I had another idea: Maybe SELinux caused that the systemd-timer was not created. I'll check this also...

> You might want to reach out to the OL8 EPEL packager.

That will be the next step ;)

**Comment by mbrde** [comment]
Actually there is a certbot-renew.timer in /usr/lib/systemd/system, which is just not enabled. Running `systemctl enable --now certbot-renew.timer` makes it work as expected. Same issue on Rocky Linux (did not check other RHEL Clones yet)

**Comment by bmw** [comment]
there's more context on this in https://github.com/certbot/certbot/issues/9931

**Comment by thiagoscodelerae** [comment]
You need to enable the certbot-renew.timer by running: `systemctl enable --now certbot-renew.timer` 

After that you can check service status: `sudo systemctl status certbot-renew.timer` - it should be active (waiting)

Finally check list timers to confirm certbot-renew.timer is there: `systemctl list-timers --all`

**Comment by killerkalamari** [comment]
Auto-renew didn't work for me and I didn't have `certbot-renew.timer`, so I hacked together a script I could use with cron:

Note: Replace the `/etc/init.d/lighttpd` commands with the appropriate commands to stop/start the web server you're using (for example: nginx).
```
sudo tee /usr/local/bin/certbot-renew >/dev/null <<EOF
#!/bin/sh
/etc/init.d/lighttpd stop
sudo certbot renew
/etc/init.d/lighttpd start
EOF
sudo chmod +x /usr/local/bin/certbot-renew
```

Edit/install your existing root crontab, or if you don't have one run these commands. This attempts to renew once a day at midnight: 
```
echo "0 0 * * * /usr/local/bin/certbot-renew" >crontab
sudo crontab crontab
```
